### ISSUE TITLE
Why does DDP gradient synchronization need to be triggered manually

### ISSUE THREAD

**Issue by liudl85** [kohya-ss/sd-scripts#2244]
Curious about these lines:
https://github.com/kohya-ss/sd-scripts/blob/3e6935a07edcb944407840ef74fcaf6fcad352f7/train_network.py#L1421-L1423

Theoretically, gradient synchronization would be automatically handled during `loss.backward()`, and in this case, it should be `accelerator.backward(loss)`.

Why is there a manual DDP gradient synchronization here? Is it redundant?

Thanks ahead

**Comment by liudl85** [comment]
Find a PR that explains this https://github.com/kohya-ss/sd-scripts/pull/989. Great work! 

I'll close this issue.
